### PR TITLE
Added retain boolean to publish params

### PR DESCRIPTION
**Pull request by TheBestJohn**
Is a solution for #120 and is a formatted, squashed commit from #121 

I think it's still a good idea to have a discussion about opinions on what was brought up in #121

### CODE REVIEW

**Comment by bbigras** [comment]
Any progress on merging this?

**Comment by tekjar** [comment]
Thanks @TheBestJohn :)

File: src/client/mod.rs [hunk 1]
```diff
@@ -77,10 +77,11 @@ impl MqttClient {
     //
     //    }
 
-    pub fn publish<S, V>(&mut self, topic: S, qos: QoS, payload: V) -> Result<(), ClientError>
+    pub fn publish<S, V, B>(&mut self, topic: S, qos: QoS, retained: B, payload: V) -> Result<(), ClientError>
```
[the remark this answers is not in the document]
Comment (TheBestJohn) [reply src/client/mod.rs]: There was no reason other than trying to follow what I saw there. That was new syntax for me :)

**Comment by TheBestJohn** [comment]
I think this also needs to bump the version number as it would be breaking to anyone using previous versions

**Comment by tekjar** [comment]
This is still a development branch. Small breaking changes are ok

**Pull request merged**